nav2_system_tests/src/system/test_system_with_obstacle_launch.py

```diff
@@ -0,0 +1,196 @@
+        os.getenv('BT_NAVIGATOR_XML'),
+    )
+
+    # Use local param file
+    launch_dir = os.path.dirname(os.path.realpath(__file__))
+    params_file = os.path.join(launch_dir, 'nav2_system_params.yaml')
+
+    # Replace the default parameter values for testing special features
+    # without having multiple params_files inside the nav2 stack
+    context = LaunchContext()
+    param_substitutions = {}
+
+    if os.getenv('ASTAR') == 'True':
+        param_substitutions.update({'use_astar': 'True'})
+
+    param_substitutions.update(
+        {'planner_server.ros__parameters.GridBased.plugin': os.getenv('PLANNER')}
+    )
+    param_substitutions.update(
+        {'controller_server.ros__parameters.FollowPath.plugin': os.getenv('CONTROLLER')}
+    )
+
+    configured_params = RewrittenYaml(
+        source_file=params_file,
+        root_key='',
+        param_rewrites=param_substitutions,
+        convert_types=True,
+    )
+
+    new_yaml = configured_params.perform(context)
+
+    cardbox_sdf = os.path.join(nav2_system_tests_dir, 'models', 'cardboard_box.sdf')
+
+    return LaunchDescription(
+        [
+            SetEnvironmentVariable('RCUTILS_LOGGING_BUFFERED_STREAM', '1'),
+            SetEnvironmentVariable('RCUTILS_LOGGING_USE_STDOUT', '1'),
+            AppendEnvironmentVariable(
+                'GZ_SIM_RESOURCE_PATH', os.path.join(sim_dir, 'models')
+            ),
+            AppendEnvironmentVariable(
+                'GZ_SIM_RESOURCE_PATH',
+                str(Path(os.path.join(sim_dir)).parent.resolve())
+            ),
+            IncludeLaunchDescription(
+                PythonLaunchDescriptionSource(
+                    os.path.join(ros_gz_sim_dir, 'launch', 'gz_sim.launch.py')
+                ),
+                launch_arguments={'gz_args': ['-r -s ', world_sdf_xacro]}.items(),
+            ),
+            IncludeLaunchDescription(
+                PythonLaunchDescriptionSource(
+                    os.path.join(sim_dir, 'launch', 'spawn_tb3.launch.py')
+                ),
+                launch_arguments={
+                    'use_sim_time': 'True',
+                    'robot_sdf': robot_sdf,
+                    'x_pose': '-2.0',
+                    'y_pose': '-0.5',
+                    'z_pose': '0.01',
+                    'roll': '0.0',
+                    'pitch': '0.0',
+                    'yaw': '0.0',
+                }.items(),
+            ),
+            Node(
+                package='ros_gz_sim',
+                executable='create',
+                output='screen',
+                arguments=[
+                    '-entity', 'cardboard_box',
+                    '-file', cardbox_sdf,
+                    '-x', '-1.0', '-y', '0.6', '-z', '0.15',
+                    '-R', '0.0', '-P', '0.0', '-Y', '0.0',]
+            ),
+            # TODO(orduno) Launch the robot state publisher instead
+            #              using a local copy of TB3 urdf file
+            Node(
```

Comment:
Are these needed after the spawn TB3 is added? I think we actually make the things work with the actual URDF now, yeah?

Reply:
I can confirm that, now that you say it. I am not sure.

Reply:
@SteveMacenski Base link and base scan is connected but base_footprint and base_link is not connected. 
```
[INFO] [1718902111.154730103] [tf2_echo]: Waiting for transform base_footprint ->  base_link: Could not find a connection between 'base_footprint' and 'base_link' because they are not part of the same tree.Tf has two or more unconnected trees.
```

Reply:
We can remove it but we have to put the robot state publisher. 
```
  Node(
                package='robot_state_publisher',
                executable='robot_state_publisher',
                name='robot_state_publisher',
                output='screen',
                parameters=[
                    {'use_sim_time': True, 'robot_description': robot_description}
                ],
            ),
```

Reply:
That sounds good!